librz/core/cmd_descs/cmd_debug.yaml

```diff
@@ -19,63 +19,186 @@ commands:
             summary: Step until an instruction matches one of the <optype>s
             args:
-            - name: optype
-              type: RZ_CMD_ARG_TYPE_STRING
-              flags: RZ_CMD_ARG_FLAG_ARRAY
+              - name: optype
+                type: RZ_CMD_ARG_TYPE_STRING
+                flags: RZ_CMD_ARG_FLAG_ARRAY
           - name: dsue
             cname: cmd_debug_step_until_esil
             summary: Step until <esil> expression matches
             args:
-            - name: esil
-              type: RZ_CMD_ARG_TYPE_STRING
+              - name: esil
+                type: RZ_CMD_ARG_TYPE_STRING
           - name: dsuf
             cname: cmd_debug_step_until_flag
             summary: Step until pc == <flag> matching name
             args:
-            - name: flag
-              type: RZ_CMD_ARG_TYPE_FLAG
+              - name: flag
+                type: RZ_CMD_ARG_TYPE_FLAG
   - name: dts
     summary: Debug trace session commands
     subcommands:
-    - name: dts+
-      summary: Start trace session
-      cname: cmd_debug_start_trace_session
-      args: []
-    - name: dts-
-      summary: Stop trace session
-      cname: cmd_debug_stop_trace_session
-      args: []
-    - name: dtst
-      cname: cmd_debug_save_trace_session
-      summary: Save trace sessions to disk
-      args:
-        - name: dir
-          type: RZ_CMD_ARG_TYPE_FILE
-    - name: dtsf
-      cname: cmd_debug_load_trace_session
-      summary: Load trace sessions to disk
-      args:
-        - name: dir
-          type: RZ_CMD_ARG_TYPE_FILE
-    - name: dtsm
-      cname: cmd_debug_list_trace_session_mmap
-      summary: List current memory map and hash
-      args: []
\ No newline at end of file
+      - name: dts+
+        summary: Start trace session
+        cname: cmd_debug_start_trace_session
+        args: []
+      - name: dts-
+        summary: Stop trace session
+        cname: cmd_debug_stop_trace_session
+        args: []
+      - name: dtst
+        cname: cmd_debug_save_trace_session
+        summary: Save trace sessions to disk
+        args:
+          - name: dir
+            type: RZ_CMD_ARG_TYPE_FILE
+      - name: dtsf
+        cname: cmd_debug_load_trace_session
+        summary: Load trace sessions to disk
+        args:
+          - name: dir
+            type: RZ_CMD_ARG_TYPE_FILE
+      - name: dtsm
+        cname: cmd_debug_list_trace_session_mmap
+        summary: List current memory map and hash
+        args: []
+  - name: dm
+    summary: Memory map commands
+    subcommands:
+      - name: dm
+        cname: cmd_debug_list_maps
+        summary: >
+          List memory maps
+        modes:
+          - RZ_OUTPUT_MODE_STANDARD
+          - RZ_OUTPUT_MODE_RIZIN
+          - RZ_OUTPUT_MODE_JSON
+          - RZ_OUTPUT_MODE_QUIET
+        args: []
+      - name: dm+
+        cname: cmd_debug_allocate_maps
+        summary: >
+          Allocate <size> bytes at current offset
+          (anywhere if offset is -1)
```

Comment:
is this really true? It seems to me `addr` is passed to the `map_alloc` method of the dbg plugin, which can do whatever it wants. For debug_native plugin, addr is passed as first argument of `mmap`, which needs NULL to allocate "anywhere", not -1.

Reply:
Oh, I am not sure about the exact working of this command, this help message was present before the refactor so I just used the same message. Should I remove the ` (anywhere if offset is -1)` part?

Reply:
For now, yes.